ocs/agents/host_manager/agent.py

```diff
@@ -303,18 +332,28 @@ def _launch_instance(self, instance):
             prot = self._get_docker_helper(instance)
         else:
             iid = instance['instance_id']
-            pyth = sys.executable
-            script = instance['agent_script']
-            if script == '__plugin__':
-                cmd = [pyth, '-m', 'ocs.agent_cli']
+            if instance.get('agent_script') is not None:
+                pyth = sys.executable
+                script = instance['agent_script']
+                if script == '__plugin__':
+                    cmd = [pyth, '-m', 'ocs.agent_cli']
+                else:
+                    cmd = [pyth, script]
+                cmd.extend([
+                    '--instance-id', iid,
+                    '--site-file', self.site_config_file,
+                    '--site-host', self.host_name,
+                    '--working-dir', self.working_dir])
+            elif instance.get('agent_exe') is not None:
+                cmd = [instance['agent_exe'], '--instance-id', self.address_root + '.' + iid,
+                       '--wamp-url', self.wamp_url, '--wamp-realm', self.wamp_realm]
```

Comment:
I understand wanting to join address_root and iid, but can you call it `--agent-address` or `--address`?  I don't like redefining `--instance-id`.

Reply:
I think I called it `--instance-id` solely out of an attempt to better fit in with the surrounding code; my agents have always taken `--address` and I later added this as an alias. So, if you prefer `--address`, so do I, and I'm happy to switch it. :)